kmath-trajectory/src/commonMain/kotlin/space/kscience/kmath/trajectory/segments/Line.kt

```diff
@@ -0,0 +1,23 @@
+package space.kscience.kmath.trajectory.segments
+
+import space.kscience.kmath.geometry.Euclidean2DSpace.distanceTo
+import space.kscience.kmath.geometry.Line2D
+import space.kscience.kmath.operations.DoubleField.pow
+import kotlin.math.PI
+import kotlin.math.atan2
+import kotlin.math.sqrt
+
+public data class LineSegment(
```

Comment:
Does not make sense to use `data class` here. Probably `value class` would be better.

Reply:
With the replacement of Line2D with the Straight segment, this is fixed